File: final_design/src/main/java/yukina/final_design_deprecated/controller/TestController.java
package yukina.final_design_deprecated.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import yukina.final_design_deprecated.domain.BusOfPlaceMapper;
import yukina.final_design_deprecated.entity.*;

import java.util.*;


@RestController
@CrossOrigin(origins = "*", maxAge = 3600)
@RequestMapping(value="/test", method=RequestMethod.GET)
public class TestController{

    private PlaceMapper placeMapper;
    @Autowired
    public void setPlaceMapper(PlaceMapper placeMapper) {
        this.placeMapper = placeMapper;
    }

    @Autowired
    private BusOfPlaceMapper buslineMapper;

    @Autowired
    private PosMapper posMapper;

    @Autowired
    private BusMapper busMapper;

    @RequestMapping("sayHello")
    @CrossOrigin(origins = "*", maxAge = 3600)
    public List<Bus> sayHello() {
        return busMapper.getBusesByID(1);
    }

    @RequestMapping("takeAJson")
    public Map<String, Object> takeAJson(@RequestParam("name") String name) {
        return new HashMap<String, Object>(){{
            put("hello", new String[]{name, "rua"});
            put("rua", new int[]{1,3,2,5,6,6});
            put("rss", new Object[]{1,3,"rus"});
            put("Happy, lucky","smile, yeah");
        }};
    }

    @RequestMapping("takeAJsonList")
    public List<Place> takeAJsonList() {
        return placeMapper.getAllPlace();
    }

}

File: final_design/src/main/kotlin/yukina/final_design/controller/TestController.kt
package yukina.final_design.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import yukina.final_design.dao.OverspeedPos
import yukina.final_design.dao.OverspeedPosMapper
import yukina.final_design.dao.PosMapper
import yukina.final_design.service.AdminService
import yukina.final_design.service.InfoGetterService
import yukina.final_design.util.getDistance
import yukina.final_design.util.parseyyyyMMdd2Date
import java.sql.Timestamp


@RestController
@CrossOrigin(origins = ["*"], maxAge = 3600)
@RequestMapping(value= ["/test"], method= [RequestMethod.GET])
class TestController @Autowired constructor(
    val infoGetterService: InfoGetterService,
    val adminService: AdminService,
    val overspeedPosMapper: OverspeedPosMapper,
    val posMapper: PosMapper
){
    @RequestMapping("foo")
    @CrossOrigin(origins = ["*"], maxAge = 3600)
    // 最短，最长，平均长度
    fun test() =
        overspeedPosMapper.getOverspeedPosesByAdcode(110101).map {
            mapOf(
                "lat" to it.lat,
                "lont" to it.lont
            )
        }

    @RequestMapping("allOverspeedPos")
    @CrossOrigin(origins = ["*"], maxAge = 3600)
    fun ttt(n:Int) =
        overspeedPosMapper.randomGetOverspeedPosesByAdcode(110101, n)
    @RequestMapping("minus")
    @CrossOrigin(origins = ["*"], maxAge = 3600)
    fun another() =
        overspeedPosMapper.getOverspeedPosesByAdcode(110101).joinToString("|") {
            """${it.bus_id},${it.lat},${it.lont},${it.line_id},${it.nodepath_id},${it.update_time}"""
        }
    @RequestMapping("change")
    @CrossOrigin(origins = ["*"], maxAge = 3600)
    fun t5gtt() =
        adminService.CHANGEPASSWD("aaaaaa")
}

